doc/Dashboard.md

```diff
@@ -0,0 +1,91 @@
+an average higher than two are green.
+
+![Screen shot of visual representation](img/dashboard-visual.png)
+
+To achieve the result shown above requires a set of "stacked" image files (one
+for each step), as well as an overlay image (in this case, the overlay image
+contains all the text).
+
+To build this example, the images used look like this:  
+![Images Used](img/dashboard-visual-instructions.png)
+
+The block was configured to use these images as follows:
+![Screen shot of visual representation rule configuration](img/dashboard-visual-config.png)
+
+Notice that step 2 has turned green due to the default rule "hueRotate 280" for
+any steps with an average above 2, and the first step has a blurred, grey
+background due to the first rule for average values less than 2.
+
+The **Visual Representation Settings** used to define the visual representation
+must be in JSON format. The supported entries are:
+
+* **`"images"`**: A list of image URLs, one per PB block, in the same order as
+  the 'blocks' list (the list of `url_name`s described above). If the images you
+  wish to use are on your computer, first upload them to the course's "Files and
+  Uploads" page. You can then find the URL for each image listed on that page.
+  Note that each URL must start with "http://" or "https://".
+    * All images listed here will be layered on top of each other, and can be
+    colorized, faded, etc. based on the average value of the student's choices
+    for the corresponding group of MCQs.
+* **`"overlay"`**: (Optional) The URL of an image to be drawn on top of the
+  layered images, with no effects applied.
+* **`"background"`**: (Optional) The URL of an image to be drawn behind the
+  layered images, with no effects applied.
+* **`"colorRules"`**: (Optional) The rules used to determine how the student's
+  average choice values affect each layer's image. Each rule is evaluated in
+  order, and the first matching rule is used. Each rule is a JSON object
+  (`{ ... }`) that may consist of any of the following optional entries.
+  * `if`: An optional expression involving `x` (the student's average choice
+    value). If this is true, then this rule will be used. If this is false, the
+    rule will not be used. If no `if` clause is set, the rule will always be
+    used, unless an earlier rule matched first.  
+    Examples:
+      - `"if": "x > 3"`
+      - `"if": "0 < x < 1"`
+      - `"if": "(x > 3) and (x < 6 or x > 100)"`
+  * `hueRotate`: If this rule matches, adjust the color of all parts of the
+    image that have some saturation. In the above example, this is used to
+    change from a turquiose color to a light green. Valid values are between 0
+    (no change) and 360. Note that this effect will *not* affect any parts of
+    the image have no saturation (i.e. are white, black, or any shade of grey).  
+    Examples:
+      - `"hueRotate": "180"`
+      - `"hueRotate": "x * 30"`
```

Comment:
Yup, I had thought of masks too, as this would be a nice approach, but I think browser compatibility prevents it - cf http://caniuse.com/#search=mask IE doesn't have any support for it, and I had searched for a polyfill without success. 

For my question, there are some conversion tools that help to convert hue to/from RGB http://www.rapidtables.com/convert/color/rgb-to-hsl.htm but to get any color from a base image we would need to also alter the saturation and luminosity values, right? Saturation is settable, but what about luminosity? I don't know much about color theory so there might already be a way I missed, but just to check.